### DISCUSSION TITLE
Add UUID column

### DISCUSSION THREAD

**Discussion by Vladimir-Va** [nocodb/nocodb#396] [Features]
Please add autogenerated UUID column.

**Comment by wingkwong** [comment]
By default, when you create a table, it only gives you auto-increment ID (the default one). However, you can add a column with autogenerated UUID with the following steps.

- Create a column with Column Type ``SpecificDBType``
- Select ``uuid`` in Type in Database
- Input your generation function in Default value box. Let's take Postgres as an example, one of the possible function is ``md5(random()::text || clock_timestamp()::text)::uuid``. The function depends on the database you are using
- Mark it as PK

After that, you can delete the default ID column. When you insert a new record, you should see a generated UUID.

**Reply by Vladimir-Va** [reply]
I try it, i can create trigger for UUID in MariaDB
![image](https://user-images.githubusercontent.com/52074398/127751541-cf404984-7af2-46c6-8bf7-e96df74552a2.png)
but can't from NocoDB GUI
![image](https://user-images.githubusercontent.com/52074398/127751598-e3284e1e-5d3a-4e89-a4b6-43feee1ea6d8.png)

Now i need add trigger as first screenshot
Add column SpecificDBType uuid varchar(36) (no have uuid type) and this work.
Hoped that we can do without PhpMyAdmin.
But if setup UUID column as PK we have error uuid field must be filled.

**Reply by Sov3rain** [reply]
Another option to create an uuid when using a postgresQL database is using a postgres function:
* Create a column with a type of `SpecificDBType`
* Select `uuid` as the type
* In the default value text box, add `gen_random_uuid()`

You're good to go.

**Comment by o1lab** [comment]
I see its locked into certain types.

What I assume we need is a template when creating tables.

**Reply by Vladimir-Va** [reply]
Or support something like create trigger field.

**Comment by PowerAppsDarren** [comment]
- Create a column with Column Type SpecificDBType
- Select uuid in Type in Database
- Input your generation function in Default value box. Let's take Postgres as an example, one of the possible function is md5(random()::text || clock_timestamp()::text)::uuid. The function depends on the database you are using
- Mark it as PK
---
Was this feature taken out? I don't see SpecificDBType anywhere. thank you!
